chickadee/processes/wps_CI.py

```diff
@@ -0,0 +1,144 @@
+                min_occurs=0,
+                max_occurs=1,
+                data_type="string",
+            ),
+            ComplexInput(
+                "out_file_overwrite",
+                "Output NetCDF File",
+                abstract="Filename to overwrite with the climate imprint outputs",
+                min_occurs=0,
+                max_occurs=1,
+                supported_formats=[FORMATS.NETCDF],
+            ),
+        ]
+
+        outputs = [
+            ComplexOutput(
+                "output",
+                "Output",
+                abstract="output netCDF file",
+                supported_formats=[FORMATS.NETCDF],
+            ),
+        ]
+
+        super(CI, self).__init__(
+            self._handler,
+            identifier="ci",
+            title="CI",
+            abstract="Climate Imprint (CI) downscaling",
+            metadata=[
+                Metadata("NetCDF processing"),
+                Metadata("Climate Data Operations"),
+            ],
+            inputs=inputs,
+            outputs=outputs,
+            store_supported=True,
+            status_supported=True,
+        )
+
+    def collect_args(self, request):
+        loglevel = request.inputs["loglevel"][0].data
+        gcm_file = request.inputs["gcm_file"][0].file
+        obs_file = request.inputs["obs_file"][0].file
+        varname = request.inputs["varname"][0].data
+        if "out_file_create" in request.inputs.keys():
+            output_file = request.inputs["out_file_create"][0].data
+        elif "out_file_overwrite" in request.inputs.keys():
+            output_file = request.inputs["out_file_overwrite"][0].file
+        else:
+            ProcessError("out_file argument not provided")
+
+        return loglevel, gcm_file, obs_file, varname, output_file
+
+    def _handler(self, request, response):
+        loglevel, gcm_file, obs_file, varname, output_file = self.collect_args(request)
+        log_handler(
+            self,
+            response,
+            "Starting Process",
+            logger,
+            log_level=loglevel,
+            process_step="start",
+        )
+
+        climdown = get_ClimDown()
+
+        log_handler(
+            self,
+            response,
+            "Processing CI downscaling",
+            logger,
+            log_level=loglevel,
+            process_step="process",
+        )
+
+        climdown.ci_netcdf_wrapper(gcm_file, obs_file, output_file, varname)
+
+        log_handler(
+            self,
+            response,
+            "Building final output",
+            logger,
+            log_level=loglevel,
+            process_step="build_output",
+        )
+
+        response.outputs["output"].file = output_file
+
+        log_handler(
+            self,
+            response,
+            "Process Complete",
+            logger,
+            log_level=loglevel,
+            process_step="complete",
+        )
+
+        return response
```

Comment:
This is such a clean cut process. The translation between R/python and the package has turned out very nice.